sound/soc/sof/ipc4-loader.c

```diff
@@ -352,9 +352,39 @@ static int sof_ipc4_load_library(struct snd_sof_dev *sdev, struct snd_sof_widget
 	return ret;
 }
 
+static int sof_ipc4_reload_libraries(struct snd_sof_dev *sdev)
+{
+	struct sof_ipc4_fw_data *ipc4_data = sdev->private;
+	int i, ret;
+
+	if (sof_ops(sdev)->load_library) {
+		dev_err(sdev->dev, "Loading module libraries not supported\n");
+		return -EINVAL;
+	}
```

Comment:
Same here, I would do the validation during probe: if fw_loader reload or load is supported, then sof_ops load_library must be.
Also, both reload_libraries _and_ load_library of fw_loader must be present if I'm not mistaken.